src/Models/Services/Utils.ts

```diff
@@ -43,12 +43,67 @@ const parser = createParser(ModelParsingOption.PermitAnyTopLevelElement);
 
 /** Parse DTDL models via model parser */
 export const parseDTDLModelsAsync = async (dtdlInterfaces: DtdlInterface[]) => {
-    const modelDict = await parser.parse(
-        dtdlInterfaces.map((dtdlInterface) => JSON.stringify(dtdlInterface))
-    );
+    let modelDict = null;
+    try {
+        modelDict = await parser.parse(
+            dtdlInterfaces.map((dtdlInterface) => JSON.stringify(dtdlInterface))
+        );
+    } catch (e) {
+        /* Some models failed to parse, we will try to remove them and parse the rest of the models */
+        logDtdlParserError(e);
+        try {
+            const parsingErrorCauses = e?._parsingErrors.map(
+                (pe) => pe?.cause
+            ) as Array<string>;
+
+            /* The parsing error cause is not a model ID, but it contains a model ID 
```

Comment:
Can't we look for a particular message instead of just seeing that it contains an id?

Reply:
See the raw error log, the only thing that contains the actual DTID is unfortunately the error cause message